modules/datagrid/72/launch/added/launch/infinispan-config.sh

```diff
@@ -231,6 +231,14 @@ function configure_infinispan_core() {
   containers="$containers ${cache_container_configuration}"
   containers="$containers $containersecurity <!-- ##INFINISPAN_CACHE## --></cache-container>"
 
+  DATAGRID_SERVICE_PROFILE="datagrid-service"
+  if [ "$PROFILE" = "$DATAGRID_SERVICE_PROFILE" ]; then
+    global_state="<global-state><overlay-configuration-storage/></global-state>"
```

Comment:
As the overlay-configuration-storage is something that could potentially be useful outside of service profiles, I think this env variable should be more generic. Furthermore, we should try to keep all of the service profile semantics in `infinispan_service_profile.sh`.

So we could create a env variable for `infinispan-config.sh`, something like `ENABLE_OVERLAY_CONFIGURATION_STORAGE`, which when = true sets `global_state="<global-state><overlay-configuration-storage/></global-state>"`

In `infinispan_service_profile.sh` when the configured service, in this case `datagrid-service` requires the storage overlay we simply set the env variable, i.e. `export ENABLE_OVERLAY_CONFIGURATION_STORAGE=true`

Reply:
@galderz This still applies

Reply:
Missed it cos it was hidden as outdated. Sounds good, let me give it a go.